smac/optimizer/acquisition.py

```diff
@@ -178,6 +179,161 @@ def _compute(self, X: np.ndarray) -> np.ndarray:
         return np.array([func._compute(X) for func in self._functions]).mean(axis=0)
 
 
+class PriorAcquisitionFunction(AbstractAcquisitionFunction):
+    r"""Weight the acquisition function with a user-defined prior over the optimum.
+
+    See "PiBO: Augmenting Acquisition Functions with User Beliefs for Bayesian Optimization" by Carl
+    Hvarfner et al. (###nolinkyet###) for further details.
+    """
+    def __init__(self, model: AbstractEPM, acquisition_function: AbstractAcquisitionFunction,\
+        decay_beta: float, prior_floor: float = 1e-12, discretize: bool = False, \
+        discrete_bins_factor: float = 10.0, **kwargs: Any):
+        """Constructor
+
+        Parameters
+        ----------
+        model : AbstractEPM
+            Models the objective function.
+        decay_beta: Decay factor on the user prior - defaults to n_iterations / 10 if not specifed
+            otherwise.
+        prior_floor : Lowest possible value of the prior, to ensure non-negativity for all values
+            in the search space. 
+        discretize : Whether to discretize (bin) the densities for continous parameters. Triggered
+            for Random Forest models and continous hyperparameters to avoid a pathological case
+            where all Random Forest randomness is removed (RF surrogates require piecewise constant
+            acquisition functions to be well-behaved)
+        discrete_bins_factor : If discretizing, the multiple on the number of allowed bins for 
+            each parameter
+
+        kwargs
+            Additional keyword arguments
+        """
+        super().__init__(model)
+        self.long_name = 'Prior Acquisition Function (%s)' % acquisition_function.__class__.__name__
+        self.acq = acquisition_function
+        self._functions = []  # type: List[AbstractAcquisitionFunction]
+        self.eta = None
+        self.hyperparameters = self.model.get_configspace().get_hyperparameters_dict()
+        self.decay_beta = decay_beta
+        self.prior_floor = prior_floor
+        self.discretize = discretize
+        if self.discretize:
+            self.discrete_bins_factor = discrete_bins_factor
+
+        # check if the acquisition function is LCB or TS - then the acquisition function values
+        # need to be rescaled to assure positiveness & correct magnitude
+        if isinstance(self.acq, IntegratedAcquisitionFunction):
+            acquisition_type = self.acq.acq
+        else:
+            acquisition_type = self.acq
+
+        self.rescale_acq = isinstance(acquisition_type, LCB) or isinstance(acquisition_type, TS)
+        self.iteration_number = 0
```

Comment:
You also need to set `self._required_updates` to fit the `acquisition_type` accordingly (see `EI.__init__` as an example)
Additionally, `eta` also needs to be included.

Reply:
@dengdifan Can you please elaborate here? I'm not sure I follow completely.

Reply:
Nevermind, I think I've got it!

Reply:
@dengdifan Sorry, I think I got confused again. Since we overwrite update in PriorAcquisitionFunction, there should be no need for self._required_updates there, right? The underlying acquisition function already has its self._required_updates, and is accordningly called as 
`        self.acq.update(**kwargs)`
in line 244. Shouldn't this be enough?